### PR TITLE
coral-schema: Generate documentation for function calls

### PR DESCRIPTION
**Pull request by sherryhli**
### Summary

Previously, field-level documentation in schemas produced by the `ViewToAvroSchemaConverter` was only available for fields that were:
* directly selected from base tables (#115)
* created from literal values (#172)
* created from aggregate function calls (#186)

This PR adds documentation for fields created from other types of function calls (i.e. UDFs and operators).

The documentation is in the form of: `"Field created in view by applying UDF / operator '<FUNCTION-NAME>' with argument(s): <LIST-OF-ARGUMENTS>"`

### Testing
There are existing unit tests covering function calls. The `.avsc` files containing expected schemas for those test cases have been updated to include the documentation field.

### CODE REVIEW

File: coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java [hunk 1]
```diff
@@ -273,20 +278,63 @@ static String getFieldName(String oldName, String suggestedNewName) {
...
+   * Given an input {@link RelNode} and the index of a field in the {@link RelNode}'s corresponding Avro schema,
+   * determine if the field with the specified index is a column from a table.
+   * @param fieldIndex the index of a field in the <code>inputRelNode</code>'s corresponding Avro schema
+   * @param inputRelNode the input {@link RelNode}
+   * @return true if the field at <code>fieldIndex</code> is a column from a table
+   */
+  private static boolean isColumn(int fieldIndex, @Nonnull RelNode inputRelNode) {
+    return !(inputRelNode instanceof LogicalAggregate)
+        || fieldIndex < ((LogicalAggregate) inputRelNode).getGroupSet().cardinality();
+  }
```
Comment (sherryhli) [inline coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java:301]: Given an input `RelNode` and a field index, how can we generalize the logic for determining whether or not the field is referring to a column from a table? I don't think this current approach is very robust.

@funcheetah @ljfgem wondering if you have any suggestions? 🙂

File: coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java [hunk 2]
```diff
@@ -273,20 +278,63 @@ static String getFieldName(String oldName, String suggestedNewName) {
...
+
+    for (RexNode rexNode : rexCall.getOperands()) {
+      SqlKind nodeKind = rexNode.getKind();
+      switch (nodeKind) {
+        case LITERAL:
+          args.add(getLiteralValueAsString((RexLiteral) rexNode));
+          break;
+        case INPUT_REF:
+          int fieldIndex = ((RexInputRef) rexNode).getIndex();
+          if (isColumn(fieldIndex, inputRelNode)) {
```
Comment (sherryhli) [inline coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java:323]: To provide more context: we only want to extract the field name if the field refers to a column, otherwise the name is not meaningful (they could be auto-assigned names such as `EXPR_3` or `_f0`).

File: coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java (hunk 1 above)
Comment (ljfgem) [reply coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java:301]: I would prefer to determine it by:
```
return !inputRelNode.getRowType().getFieldNames().get(fieldIndex).contains("$");
```
`$` wouldn't appear in a normal column name, but would appear in the auto-assigned column name in the row type.

Comment (sherryhli) [reply coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java:301]: Thank you, I'll give this a try!

Comment (sherryhli) [reply coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java:301]: @ljfgem This seems to work for all current test cases, but it looks like from Hive 0.13 onwards, `$` can be a valid character in a column name.

From Apache Hive [docs](https://cwiki.apache.org/confluence/display/hive/languagemanual+ddl#LanguageManualDDL-CreateTable):
>* In Hive 0.12 and earlier, only alphanumeric and underscore characters are allowed in table and column names.
>* In Hive 0.13 and later, column names can contain any Unicode character (see HIVE-6013), however, dot (.) and colon (:) yield errors on querying, so they are disallowed in Hive 1.2.0 (see HIVE-10120). Any column name that is specified within backticks (`) is treated literally. Within a backtick string, use double backticks (``) to represent a backtick character. Backtick quotation also enables the use of reserved keywords for table and column identifiers.

Comment (ljfgem) [reply coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java:301]: Hmmm, then I am good with the current logic for determining it.

Comment (sherryhli) [reply coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java:301]: Sounds good. Are there any special cases other than `LogicalAggregate` that I should account for?

Comment (ljfgem) [reply coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java:301]: Nothing on top of my mind at present, we can add if we encounter a corner case in the future.

**Review by ljfgem — APPROVED** [review]
LGTM, thanks @sherryhli for this great work!
i-test passed.

File: coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java (hunk 1 above)
Comment (funcheetah) [reply coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java:301]: I'm okay with this approach to determine whether it is a column.

**Review by funcheetah — APPROVED** [review]
Nice work! Thanks for your contribution to Coral @sherryhli !
